dataset/src/main/scala/frameless/TypedColumn.scala

```diff
@@ -411,80 +439,168 @@ sealed class TypedColumn[T, U](
+  def ^(u: TC[U])(implicit n: CatalystBitwise[U]): TC[U] = bitwiseXOR(u)
 
   /** Casts the column to a different type.
     * {{{
     *   df.select(df('a).cast[Int])
     * }}}
     */
-  def cast[A: TypedEncoder](implicit c: CatalystCast[U, A]): TypedColumn[T, A] =
-    self.untyped.cast(TypedEncoder[A].catalystRepr).typed
+  def cast[A: TypedEncoder](implicit c: CatalystCast[U, A]): TC[A] =
+    mkTyped(self.untyped.cast(TypedEncoder[A].catalystRepr))
 
   /** Contains test.
     * {{{
     *   df.filter ( df.col('a).contains("foo") )
     * }}}
     */
-  def contains(other: String)(implicit ev: U =:= String): TypedColumn[T, Boolean] =
-    self.untyped.contains(other).typed
+  def contains(other: String)(implicit ev: U =:= String): TC[Boolean] =
+    mkTyped(self.untyped.contains(other))
 
   /** Contains test.
     * {{{
     *   df.filter ( df.col('a).contains(df.col('b) )
     * }}}
     */
-  def contains(other: TypedColumn[T, U])(implicit ev: U =:= String): TypedColumn[T, Boolean] =
-    self.untyped.contains(other.untyped).typed
+  def contains(other: TC[U])(implicit ev: U =:= String): TC[Boolean] =
+    mkTyped(self.untyped.contains(other.untyped))
 
   /** Boolean AND.
     * {{{
     *   df.filter ( (df.col('a) === 1).and(df.col('b) > 5) )
     * }}}
     */
-  def and(other: TypedColumn[T, Boolean]): TypedColumn[T, Boolean] =
-    self.untyped.and(other.untyped).typed
+  def and(other: TC[Boolean]): TC[Boolean] =
+    mkTyped(self.untyped.and(other.untyped))
 
   /** Boolean AND.
     * {{{
     *   df.filter ( df.col('a) === 1 && df.col('b) > 5)
     * }}}
     */
-  def && (other: TypedColumn[T, Boolean]): TypedColumn[T, Boolean] =
-    and(other)
+  def && (other: TC[Boolean]): TC[Boolean] = and(other)
 
   /** Boolean OR.
     * {{{
     *   df.filter ( (df.col('a) === 1).or(df.col('b) > 5) )
     * }}}
     */
-  def or(other: TypedColumn[T, Boolean]): TypedColumn[T, Boolean] =
-    self.untyped.or(other.untyped).typed
+  def or(other: TC[Boolean]): TC[Boolean] =
+    mkTyped(self.untyped.or(other.untyped))
 
   /** Boolean OR.
     * {{{
     *   df.filter ( df.col('a) === 1 || df.col('b) > 5)
     * }}}
     */
-  def || (other: TypedColumn[T, Boolean]): TypedColumn[T, Boolean] =
-    or(other)
-}
+  def || (other: TC[Boolean]): TC[Boolean] = or(other)
 
-/** Expression used in `groupBy`-like constructions.
-  *
-  * @tparam T type of dataset
-  * @tparam U type of column for `groupBy`
-  */
-sealed class TypedAggregate[T, U](val expr: Expression)(
-  implicit
-  val uencoder: TypedEncoder[U]
-) extends UntypedExpression[T] {
+  /**
+    * Less than.
+    * {{{
+    *   // The following selects people younger than the maxAge column.
+    *   df.select( df('age) < df('maxAge) )
+    * }}}
+    *
+    * @param u another column of the same type
+    * apache/spark
+    */
+  def <(u: TypedColumn[T, U])(implicit canOrder: CatalystOrdered[U]): TC[Boolean] =
```

Comment:
Why is `<` from `TypedColumn` to `TC`, whereas `or` is from `TC` to `TC`?

Reply:
Good catch, missed this one.

Reply:
I think the next 3 functions also have the same typo